### ISSUE TITLE
EDMEulerScheduler fails when fineturning CosXL_edit model.

### ISSUE THREAD

**Issue by leeruibin** [huggingface/diffusers#8356]
When I try to fineturning CosXL model. As they stated in [CosXL](https://huggingface.co/stabilityai/cosxl), CosXL employ Cosine-Continuous EDM VPred scheduler to turn the model. I follow the instruction of [train_dreambooth_lora_sdxl](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora_sdxl.py) to fineturn CosXL model as there are official implementation of edm_style_training by diffuser. However, the model perfermance will become terrible after I try to fineturn it.

I have tried EDMEulerScheduler and DDPMScheduler but none of them seems work well. I have on idea whether the EDMEulerScheduler is exactly the Cosine-Continuous EDM VPred scheduler mentioned by StablilityAI or not. They only release the model parameters and give a very simple statement. 

I would be grateful if you could give me some suggestions to make the fineturninig progress work normally.

In my setting, I can edit image successfully when I use instruction like "change the cup to red" before fineturning. However, when I try to fineturn the CosXL_edit model with [fusing/instructpix2pix-1000-samples](https://huggingface.co/datasets/fusing/instructpix2pix-1000-samples) dataset as a toy experiment, the model becomes weird. The loss can decrease normally but the editing fails and the output image contains many artifacts.

My training code is in [here](https://github.com/leeruibin/Attempt_CosXL_fineturning/blob/main/train_CosXLEdit_customized_data.py). Most of the codes are copied from [train_dreambooth_lora_sdxl](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora_sdxl.py) and [cosxl](https://huggingface.co/spaces/multimodalart/cosxl/tree/main) and here are my fineturning results. 

![335271679-92411946-c491-4c23-96b8-8ff51eea3912](https://github.com/huggingface/diffusers/assets/26408630/3f237be3-0daf-4d26-aa6b-a87ca1f47ad7)

I think this is caused by the wrong setting of the training scheduler. Currently, I follow the setting of [train_dreambooth_lora_sdxl](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora_sdxl.py) to set the training scheduler to EDMEulerScheduler, I am not sure whether this is the Cosine-Continuous EDM VPred schedule mentioned by stabilityAI.

The inference setting of the fineturned model is the same as the first line. I use inference step 20, CFG=7, no negative prompt.

The earlier checkpoints also fail. This is the evaluation result after 3K steps. The training batch size is 12.

Thanks a lot!
![335283134-ddf15f54-9364-4db5-9edc-af47f34731bb](https://github.com/huggingface/diffusers/assets/26408630/82fc78d6-5205-487d-a5d7-9bf7c6099021)

_Originally posted by @leeruibin in https://github.com/huggingface/diffusers/issues/8281#issuecomment-2139150466_

**Comment by sayakpaul** [comment]
Yeah the script doesn't support the scheduling and training objective used by the CosXL model. I think you will have to modify it accordingly so that it fits the same objective. 

To me, this also feels more like a discussion rather than an issue in the first place.

**Comment by leeruibin** [comment]
Thanks for you kind reply. You are right, I should have open a discussion session rather than an issue session.

If you or other viewers have any idea about how to set the right scheduling and trainig objective for fineturning CosXL model later on. I would appreciate it.

Thanks!!
